src/main/java/htsjdk/samtools/BinaryCigarCodec.java

```diff
@@ -46,7 +46,7 @@
         int binaryCigarLength = 0;
         for (int i = 0; i < cigar.numCigarElements(); ++i) {
             final CigarElement cigarElement = cigar.getCigarElement(i);
-            final int op = CigarOperator.enumToBinary(cigarElement.getOperator());
+            final int op = cigarElement.getOperator().bamCode();
```

Comment:
Yes!  I've looked at this and been annoyed by the fact that this is computed rather than hardcoded before.